fhir-path/src/main/java/com/ibm/fhir/path/patch/FHIRPathPatch.java

```diff
@@ -17,33 +16,189 @@
+    @Override
+    public <T extends Resource> T apply(T resource) throws FHIRPatchException {
+        for (FHIRPathPatchOperation fhirPathPatchOperation : operations) {
+            resource = fhirPathPatchOperation.apply(resource);
+        }
+        return resource;
+    }
+    
     /**
+     * Convert the FHIRPathPatch to a FHIR Parameters resource
+     */
+    public Parameters toParameters() {
+        Parameters.Builder builder = Parameters.builder();
+        for (FHIRPathPatchOperation operation : operations) {
+            builder.parameter(operation.toParameter());
+        }
+        return builder.build();
+    }
+    
+    public Builder toBuilder() {
+        return new Builder().from(this);
+    }
+    
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    public static class Builder { 
+        private List<FHIRPathPatchOperation> operations = new ArrayList<>(3);
+        
+        private Builder() {
+            // hidden constructor
+        }
+    
+        public Builder add(String path, String elementName, Element element) {
+            operations.add(new FHIRPathPatchAdd(path, elementName, element));
+            return this;
+        }
+        
+        public Builder delete(String path) {
+            operations.add(new FHIRPathPatchDelete(path));
+            return this;
+        }
+        
+        public Builder insert(String path, Element element, Integer index) {
+            operations.add(new FHIRPathPatchInsert(path, element, index));
+            return this;
+        }
+        
+        public Builder move(String path, Integer source, Integer destination) {
+            operations.add(new FHIRPathPatchMove(path, source, destination));
+            return this;
+        }
+        
+        public Builder replace(String path, Element element) {
+            operations.add(new FHIRPathPatchReplace(path, element));
+            return this;
+        }
+        
+        /**
+         * Build the {@link FHIRPathPatch}
+         * 
+         * @return
+         *     An immutable object of type {@link FHIRPathPatch}
+         */
+        public FHIRPathPatch build() {
+            return new FHIRPathPatch(this);
+        }
+
+        protected Builder from(FHIRPathPatch patch) {
+            operations.addAll(patch.operations);
+            return this;
+        }
+    }
+    
+    /**
+     * Parse a FHIRPathPatch from a FHIR Parameters resource
+     * 
      * @throws IllegalArgumentException if the Parameters object does not satisfy the requirements of a FHIRPath Patch
      */
-    public FHIRPathPatch(Parameters params) {
+    public static FHIRPathPatch from(Parameters params) {
         Objects.requireNonNull(params);
-        operations = new ArrayList<FHIRPathPatchOperation>(2);
+        Builder builder = FHIRPathPatch.builder();
+        
         for (Parameter param : params.getParameter()) {
             if (!"operation".equals(param.getName().getValue())) {
```

Comment:
Is there a constant for this defined somewhere?

Reply:
yeah, good catch
```suggestion
            if (!FHIRPathPatchOperation.OPERATION.equals(param.getName().getValue())) {
```